### ISSUE TITLE
[JENKINS-43411] Evaluate yarn

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/blueocean-plugin#4020]
<p>Given the increasing tempo of build failures to to npm, the challenges with shrinkwrap and the inability to use mirrors, it may be time to evaluate the use of yarn as an alternative.</p>

<p>Approach</p>
<ul class="alternate" type="square">
	<li>General things to validate
	<ul class="alternate" type="square">
		<li>Ensure "run scripts" still work with yarn</li>
		<li>maven-frontend-plugin supports yarn natively, so we just need to switch this over.</li>
		<li>Do local installs work nicely? (for rapid development). Can we obviate or at least better integrate w/ slink so rapid dev is more seamless? Check out "yarn link"</li>
	</ul>
	</li>
	<li>JDL Next
	<ul class="alternate" type="square">
		<li>Need to determine if/how a Yarn-published module is able to freeze its dependencies when pulled in via npm (since there is no shrink, probably not, but maybe there's some magic)</li>
		<li>Publish some beta JDL's and then pull them in via a blueocean-plugin PR to ensure ATH passes and things work, etc</li>
	</ul>
	</li>
	<li>blueocean-core-js
	<ul class="alternate" type="square">
		<li>Do the same as above</li>
		<li>At this point perhaps we add JDL as a peerDep and avoid the need to republish core-js each time new JDL is published (large pain point)</li>
	</ul>
	</li>
	<li>blueocean-web, dashboard and personalization
	<ul class="alternate" type="square">
		<li>Port these over to yarn as well</li>
	</ul>
	</li>
	<li>Test with Artifactory</li>
</ul>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=michaelneale">michaelneale</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-43411" target="_blank">Evaluate yarn</a></i></summary>
<i><ul>
<li><b>status</b>: Resolved
<li><b>priority</b>: Major
<li><b>component(s)</b>: blueocean-plugin
<li><b>label(s)</b>: technical-debt
<li><b>resolution</b>: Postponed
<li><b>resolved</b>: 2018-02-05T05:34:08+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 3
<li><b>imported</b>: 2025-12-01
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>Given the increasing tempo of build failures to to npm, the challenges with shrinkwrap and the inability to use mirrors, it may be time to evaluate the use of yarn as an alternative.</p>

<p>Approach</p>
<ul class="alternate" type="square">
	<li>General things to validate
	<ul class="alternate" type="square">
		<li>Ensure "run scripts" still work with yarn</li>
		<li>maven-frontend-plugin supports yarn natively, so we just need to switch this over.</li>
		<li>Do local installs work nicely? (for rapid development). Can we obviate or at least better integrate w/ slink so rapid dev is more seamless? Check out "yarn link"</li>
	</ul>
	</li>
	<li>JDL Next
	<ul class="alternate" type="square">
		<li>Need to determine if/how a Yarn-published module is able to freeze its dependencies when pulled in via npm (since there is no shrink, probably not, but maybe there's some magic)</li>
		<li>Publish some beta JDL's and then pull them in via a blueocean-plugin PR to ensure ATH passes and things work, etc</li>
	</ul>
	</li>
	<li>blueocean-core-js
	<ul class="alternate" type="square">
		<li>Do the same as above</li>
		<li>At this point perhaps we add JDL as a peerDep and avoid the need to republish core-js each time new JDL is published (large pain point)</li>
	</ul>
	</li>
	<li>blueocean-web, dashboard and personalization
	<ul class="alternate" type="square">
		<li>Port these over to yarn as well</li>
	</ul>
	</li>
	<li>Test with Artifactory</li>
</ul>

</pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
- _Remote links associated with this issue:_


  - [plugin-pom PR 57](https://github.com/jenkinsci/plugin-pom/pull/57)
  - [blueocean-plugin PR 981](https://github.com/jenkinsci/blueocean-plugin/pull/981)

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jglick.png" title="jglick's avatar" /> jglick:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-43411?focusedId=296155&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-296155">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Is this intended to solve part/all of INFRA-1139, or is it related to some extent?</p>
</pre>
</details>

</ul>
</details>

<p>Is this intended to solve part/all of INFRA-1139, or is it related to some extent?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> michaelneale:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-43411?focusedId=296157&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-296157">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jglick" class="user-hover" rel="jglick">jglick</a> partially yes. That specific issue that happened recently wasn't the npm registry, more a repo of binaries (still have to work out how to mirror that), but the more common case is this. </p>

<p> </p>

<p>Yarn doesn't specifically change things to allow the jenkins npm registry to be used but it is one of the side benefits. </p>

<p> </p>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=cliffmeyers" class="user-hover" rel="cliffmeyers">cliffmeyers</a> this is in review - what is best way to test this? and IIUC the plugin-pom change wont' make it use the artifactory proxy automatically? </p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jglick" class="user-hover" rel="jglick">jglick</a> partially yes. That specific issue that happened recently wasn't the npm registry, more a repo of binaries (still have to work out how to mirror that), but the more common case is this. </p>

<p> </p>

<p>Yarn doesn't specifically change things to allow the jenkins npm registry to be used but it is one of the side benefits. </p>

<p> </p>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=cliffmeyers" class="user-hover" rel="cliffmeyers">cliffmeyers</a> this is in review - what is best way to test this? and IIUC the plugin-pom change wont' make it use the artifactory proxy automatically? </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/cliffmeyers.png" title="cliffmeyers's avatar" /> cliffmeyers:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-43411?focusedId=296273&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-296273">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I did not put this in review just because there will be multiple PR's and more testing that needs to be done in CI. I suppose it's semantics anyways.</p>
</pre>
</details>

</ul>
</details>

<p>I did not put this in review just because there will be multiple PR's and more testing that needs to be done in CI. I suppose it's semantics anyways.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/cliffmeyers.png" title="cliffmeyers's avatar" /> cliffmeyers:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-43411?focusedId=296274&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-296274">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>First PR that adds yarn support to plugin-pom is here: <a href="https://github.com/jenkinsci/plugin-pom/pull/57" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/plugin-pom/pull/57</a></p>
</pre>
</details>

</ul>
</details>

<p>First PR that adds yarn support to plugin-pom is here: <a href="https://github.com/jenkinsci/plugin-pom/pull/57" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/plugin-pom/pull/57</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jglick.png" title="jglick's avatar" /> jglick:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-43411?focusedId=296283&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-296283">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Yup, already linked above.</p>
</pre>
</details>

</ul>
</details>

<p>Yup, already linked above.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/cliffmeyers.png" title="cliffmeyers's avatar" /> cliffmeyers:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-43411?focusedId=296401&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-296401">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Second PR that uses yarn to build blue ocean modules is here: <a href="https://github.com/jenkinsci/blueocean-plugin/pull/981" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/blueocean-plugin/pull/981</a></p>
</pre>
</details>

</ul>
</details>

<p>Second PR that uses yarn to build blue ocean modules is here: <a href="https://github.com/jenkinsci/blueocean-plugin/pull/981" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/blueocean-plugin/pull/981</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/cliffmeyers.png" title="cliffmeyers's avatar" /> cliffmeyers:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-43411?focusedId=300565&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-300565">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>This is on hold until new extensibility system lands. Both from a conflict standpoint, but also hopefully because a lot of our need to publish to npm registry may be obviated by it, and there are some pitfalls with publishing to npm/yarn registry when using a custom repo / mirror.</p>
</pre>
</details>

</ul>
</details>

<p>This is on hold until new extensibility system lands. Both from a conflict standpoint, but also hopefully because a lot of our need to publish to npm registry may be obviated by it, and there are some pitfalls with publishing to npm/yarn registry when using a custom repo / mirror.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jamesdumay.png" title="jamesdumay's avatar" /> jamesdumay:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-43411?focusedId=303725&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-303725">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Move this to backlog unit we can get Viveks time to unblock</p>
</pre>
</details>

</ul>
</details>

<p>Move this to backlog unit we can get Viveks time to unblock</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jamesdumay.png" title="jamesdumay's avatar" /> jamesdumay:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-43411?focusedId=327556&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-327556">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Postponing. Looks like NPM is less crappy nowdays?</p>
</pre>
</details>

</ul>
</details>

<p>Postponing. Looks like NPM is less crappy nowdays?</p>
